pageserver/src/walreceiver.rs

```diff
@@ -295,6 +296,9 @@ fn walreceiver_main(
                     caught_up = true;
                 }
 
+                // Release memory if needed
+                layered_repository::evict_layer_if_needed()?;
+
```

Comment:
Would be cool if we could avoid reading the WAL record before ensuring we had enough space